File: src/main/java/com/testproject/demo/student/StudentService.java
package com.testproject.demo.student;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class StudentService {
    private final StudentRepository studentRepository;

    @Autowired
    public StudentService(StudentRepository studentRepository) {
        this.studentRepository = studentRepository;
    }

    List<Student> getStudents() {
        System.out.println("Serving students...");
        return studentRepository.findAll();
    }


    public void addStudent(Student student) {
        Optional<Student> studentByEmail = studentRepository.findStudentByEmail(student.getEmail());
        if (studentByEmail.isPresent()) {
            throw new IllegalStateException("Email " + student.getEmail() + " is taken.");
        }
        System.out.println("Adding student: " + student);
        studentRepository.save(student);
    }
}

File: src/main/java/com/testproject/demo/student/StudentService.kt
package com.testproject.demo.student

import org.springframework.stereotype.Service
import java.time.LocalDate
import java.time.Month

@Service
class StudentService {
    fun getStudents(): List<Student> {
        return listOf(
            Student(1L, "Marian", "email@gmail.com", LocalDate.of(2001, Month.FEBRUARY, 10), 22),
            Student(1L, "John", "email2@gmail.com", LocalDate.of(2000, Month.DECEMBER, 18), 23)
        )
    }
}